core/src/main/java/org/apache/brooklyn/core/objs/AbstractConfigurationSupportInternal.java

```diff
@@ -138,18 +131,14 @@ public T call() {
         // getRaw returns Maybe(val) if the key was explicitly set (where val can be null)
         // or Absent if the config key was unset.
         Object unresolved = getRaw(key).or(key.getDefaultValue());
-        final Object marker = new Object();
-        // Give tasks a short grace period to resolve.
-        Object resolved = Tasks.resolving(unresolved)
+        Maybe<Object> resolved = Tasks.resolving(unresolved)
                 .as(Object.class)
-                .defaultValue(marker)
                 .immediately(true)
                 .deep(true)
                 .context(getContext())
-                .get();
-        return (resolved != marker)
-                ? TypeCoercions.tryCoerce(resolved, key.getTypeToken())
-                        : Maybe.<T>absent();
+                .getMaybe();
+        if (resolved.isAbsent()) return Maybe.Absent.<T>castAbsent(resolved);
+        return TypeCoercions.tryCoerce(resolved.get(), key.getTypeToken());
```

Comment:
Now that there's no marker object you can rely on `ValueResolver`'s coercion directly (by passing in the expected type).